vault/core_locking.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package vault
```

Comment:
I'm curious if you considered putting this in the locking helper package instead of the `vault` package. Not a blocker by any means, but seems like a decent place for this to live.

Reply:
I hadn't thought of it at all but it does make sense and now that it's complete, I can see that nothing requires it to be in the `vault` package.